src/security/Handshake.cc

```diff
@@ -550,23 +551,26 @@ Security::HandshakeParser::ParseCertificate(const SBuf &raw, Security::CertPoint
     pCert = nullptr;
     debugs(83, 2, "TLS parsing is not supported without OpenSSL. " << raw);
 #endif
+    return pCert;
```

Comment:
> maybe we do not need to fix ParseCertificate to return a CertPointer

You do want a smart pointer inside ParseCertificate() because that method may throw after allocating the object.